src/librbd/api/Mirror.cc

```diff
@@ -3263,12 +3217,22 @@ int Mirror<I>::group_promote(IoCtx& group_ioctx, const char *group_name,
   }
 
   int ret_code = 0;
-  std::vector<uint64_t> snap_ids(image_ctxs.size(), CEPH_NOSNAP);
+  bool group_snap_rollback = false;
   std::vector<C_SaferCond*> on_finishes(image_ctxs.size(), nullptr);
+  if (snap_ids.empty()) {
+    snap_ids.resize(image_ctxs.size(), CEPH_NOSNAP);
+  } else {
+    group_snap_rollback = true;
+  }
 
   for (size_t i = 0; i < image_ctxs.size(); i++) {
     C_SaferCond* on_finish = new C_SaferCond;
-    image_promote(image_ctxs[i], group_snap_id, force, &snap_ids[i], on_finish);
+    ldout(cct, 10) << "image promotion: image_name: " << image_ctxs[i]->name.c_str()
+                   << ", group_snap_id: " << group_snap_id
+                   << ", rollback_snap_id: " << snap_ids[i]
+                   << dendl;
+    image_promote(image_ctxs[i], group_snap_id, group_snap_rollback, force,
+                  &snap_ids[i], on_finish);
```

Comment:
Passing the rollback_snap_id as an argument that is used to receive the ID generated by the `PromoteRequest` state machine doesn't seems right to me. See that its data type is `uint64 *`, a pointer of uint64_t, which indicates that it's used to receive an ID and it's not to be used to set an ID.